### PR TITLE
Clean log

### PR DESCRIPTION
**Pull request by adz**

### CODE REVIEW

**Comment by mnoack** [comment]
:+1:

File: lib/background_worker.rb [hunk 1]
```diff
@@ -1,15 +1,19 @@
-module BackgroundWorker
-  VERSION = '0.0.1'
-
-  require 'background_worker/base'
-  require 'background_worker/persistent_state'
+require 'background_worker/version'
```
Comment (AlessandroBerardi) [inline lib/background_worker.rb]: Hound, have a chat to Rubocop and find some agreement, will you? Thanks!

File: lib/background_worker.rb [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+require 'background_worker/base'
```
Comment (AlessandroBerardi) [inline lib/background_worker.rb]: Hound, have a chat to Rubocop and find some agreement, will you? Thanks!

File: lib/background_worker.rb [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+require 'background_worker/persistent_state'
```
Comment (AlessandroBerardi) [inline lib/background_worker.rb]: Hound, have a chat to Rubocop and find some agreement, will you? Thanks!

File: lib/background_worker.rb [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
   def self.after_exception(&block)
     if block
       @@after_exception = block
-    else
-      @@after_exception if defined?(@@after_exception)
+      return
+    end
+
+    @@after_exception ||= lambda  do |e|
+      logger.error '** No after_exception handler installed **'
```
Comment (AlessandroBerardi) [inline lib/background_worker.rb]: Hound, have a chat to Rubocop and find some agreement, will you? Thanks!

File: lib/background_worker/base.rb [hunk 5]
```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
...
-      end
-
-
       # This method is called by the job runner
       #
       # It will just call your preferred method in the worker.
-      def perform(method_name, options={})
-        rescue_reconnect if defined?(Resque)
-        ActiveRecord::Base.verify_active_connections!
+      def perform(method_name, options = {})
```
Comment (AlessandroBerardi) [inline lib/background_worker/base.rb]: Yes it is.

File: lib/background_worker/base.rb [hunk 6]
```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
...
+        fail ArgumentError, ":uid is required: Options given were: #{options.inspect}" if options['uid'].blank?
```
Comment (AlessandroBerardi) [inline lib/background_worker/base.rb]: Ok for the line being too long, but for the rest, Hound, have a chat to Rubocop and find some agreement, will you? Thanks!

File: lib/background_worker/base.rb [hunk 7]
```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
...
+        BackgroundWorker.verify_active_connections!
 
-        raise ArgumentError, ":uid is required: Options given were: #{options.inspect}" if options['uid'].blank?
-        setup_logger(method_name)
-        raise ArgumentError, ":current_user_id is required: Options given were: #{options.inspect}" if options['current_user_id'].blank?
+        # Special 'user' handling
+        fail ArgumentError, ":current_user_id is required: Options given were: #{options.inspect}" if options['current_user_id'].blank?
```
Comment (AlessandroBerardi) [inline lib/background_worker/base.rb]: Ok for the line being too long, but for the rest, Hound, have a chat to Rubocop and find some agreement, will you? Thanks!

File: lib/background_worker/base.rb [hunk 8]
```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
...
 
-        worker.log "returned data : #{returned_data.inspect}"
-        worker.log "completed : #{worker.state.completed}"
-        if !worker.state.completed
-          worker.log "data : #{worker.state.data}"
-          worker.state.data.merge!(hasherize_opts(returned_data, :key_if_not_hash => :result))
-          worker.log "data : #{worker.state.data}"
+        # If not explicitly completed, set completed now
+        unless worker.state.completed
+          worker.state.data.merge!(hasherize_opts(returned_data, key_if_not_hash: :result))
```
Comment (AlessandroBerardi) [inline lib/background_worker/base.rb]: The limit should be 100

File: lib/background_worker/base.rb [hunk 9]
```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
...
         end
 
       rescue Exception => e
-        puts "ERROR: #{e} \n #{e.backtrace.join("\n")}"
-        if logger
-          log("#{options['uid']}: Exception: #{e}", :error)
-          log("#{options['uid']}: Invalid Record: #{e.record.errors.full_messages.to_sentence}", :error) if e.is_a?(ActiveRecord::RecordInvalid)
-          log("#{options['uid']}: #{e.backtrace.join("\n")}", :error)
+        if worker
+          worker.log("Implicit failure: Exception occurred: #{e}", severity: :error)
```
Comment (AlessandroBerardi) [inline lib/background_worker/base.rb]: The limit should be 100

File: lib/background_worker.rb (hunk 1 above)
Comment (msmirnoff) [inline lib/background_worker.rb]: I believe single vs double is a configurable setting.

File: lib/background_worker.rb (hunk 2 above)
Comment (msmirnoff) [inline lib/background_worker.rb]: I believe single vs double is a configurable setting.

File: lib/background_worker.rb (hunk 3 above)
Comment (msmirnoff) [inline lib/background_worker.rb]: I believe single vs double is a configurable setting.

File: lib/background_worker.rb (hunk 4 above)
Comment (msmirnoff) [inline lib/background_worker.rb]: I believe single vs double is a configurable setting.

File: lib/background_worker/base.rb (hunk 5 above)
Comment (msmirnoff) [inline lib/background_worker/base.rb]: The method is trailing-if addicted. This is more branchy than a small tree.

File: lib/background_worker/base.rb (hunk 6 above)
Comment (msmirnoff) [inline lib/background_worker/base.rb]: I believe single vs double is a configurable setting.

File: lib/background_worker/base.rb (hunk 7 above)
Comment (msmirnoff) [inline lib/background_worker/base.rb]: I believe single vs double is a configurable setting.

**Comment by msmirnoff** [comment]
:+1:

**Comment by msmirnoff** [comment]
:+1:

**Pull request merged**